### ISSUE TITLE
Remove `get_items` operations since we already have `iter_items`

### ISSUE THREAD

**Issue by itziakos** [simphony/simphony-common#193]
Hi all,

As discussed in the meeting we need to decide on removing the `get_items` methods from the containers since `iter_items` when passing a list of uuid values gives similar results.

If this change is approved then the following operation 
```python
items = self.get_items(uids)
```
will become
```python
items = [item of item in self.iter_items(uids)]
```

As far as I can tell, this is the only change that the end user will need to do. What is you opinion?

**Comment by nathanfranklin** [comment]
>As discussed in the meeting we need to decide on removing the get_items methods from the containers since iter_items when passing a list of uuid values gives similar results.

I just got a quick question to clarify  things as I missed the meeting (sorry about that). I know in #178, having a `get_items` instead of a `get_item` was discussed.  Those weren't added yet, right?  Is this issue referring to removing the 'get_items', the existing "singular" version `get_item`, or both?

**Comment by itziakos** [comment]
> I just got a quick question to clarify things as I missed the meeting (sorry about that). I know in #178, having a get_items instead of a get_item was discussed. Those weren't added yet, right? Is this issue referring to removing the 'get_items', the existing "singular" version get_item, or both?

The discussion was specific to the plural version of `get_items` and the fact that in the new api it practically becomes an list comprehension using `iter_items`. The `get_item` vs `get_items` is not part of this discussion.

**Comment by nathanfranklin** [comment]
Cool.  Thanks for the feedback.  I prefer just having `iter_items` and not having `get_items`

**Comment by roigcarlo** [comment]
I believe that it would be interesting to have at least one example to check whether it is usefull to have ```get_items()```  or not, but my opinion the plural variant ```get_items()``` should not exist as it almost duplicates the functionality of ```iter_items()```. 

The only reason I might think for ```get_items()``` to exist is to keep symmetry with the rest of the ```add_items()```, ```update_items()```, etc... which it is true that it might be easier for the user.

Said that I also think that one of the perks of using python is to be able to type things like:
```python
items = [item of item in self.iter_items(uids)] 
```

Apart from that,
> The get_item vs get_items is not part of this discussion.

Maybe we should discuss about this. Again this is just a use case of ```iter_items()``` where only one item is requested. We use it internally to implement ```iter_items()``` in fact and I think is good to have this method in the API, but it may be worth to consider if someone is going to ever use this except for small test examples. I cannot think of any situation where I would need to use ```get_item()``` in DEM and CFD context except maybe for debug purposes and if it where the case it could be done with iterators.

In conclusion, my point is that if the only reason ( in my opinion ) to add ```get_items()``` is to keep a kind of logical API but ```get_item()``` is not going to be used in the first place, then is not worth to add it to the API.

**Comment by itziakos** [comment]
> Maybe we should discuss about this.

My point was that  `get_item` vs `get_items` should not be discussed in this issue to avoid confusion. Please open a separate issue about the singular and plural versions off  get, add and update.

**Comment by itziakos** [comment]
> In conclusion, my point is that if the only reason ( in my opinion ) to add get_items() is to keep a kind of logical API but get_item() is not going to be used in the first place, then is not worth to add it to the API.

I am in favor of this, but as you mention the library users want a `get_items` we can  reintroduce it at a later stage as a special use of `iter_items`. It could be part of the abstract classes (yes this is possible in python :P)
